static/js/modules/filters.js

```diff
@@ -1,217 +1,122 @@
-  $('body').addClass('is-showing-filters');
-  $('.filters').addClass('is-open');
-  $('#filter-toggle').addClass('is-active')
-    .find('.filters__toggle__text').html('Hide filters');
-  accessibility.restoreTabindex($('#category-filters'));
-  open = true;
-};
 
-var closeFilterPanel = function() {
-  $('body').removeClass('is-showing-filters');
-  $('.filters.is-open').removeClass('is-open');
-  $('#filter-toggle').removeClass('is-active')
-    .find('.filters__toggle__text').html('Show filters');
-  $('#results tr:first-child').focus();
-  accessibility.removeTabindex($('#category-filters'));
-  open = false;
-};
+// Hack: Append jQuery to `window` for use by legacy libraries
+window.$ = window.jQuery = $;
+
+var typeahead = require('fec-style/js/typeahead');
+var typeaheadFilter = require('fec-style/js/typeahead-filter');
+
+var helpers = require('./helpers');
 
-// Keep in sync with styles/grid-settings.scss.
-// TODO find better way to sync with scss.
-if ($('body').width() > 768) {
-  open = true;
-  openFilterPanel();
+var KEYCODE_ENTER = 13;
+
+function prepareValue($elm, value) {
+  return $elm.attr('type') === 'checkbox' ?
+    helpers.ensureArray(value) :
+    value;
 }
 
-$('#filter-toggle').click(function(){
-  if ( open === true ) {
-    closeFilterPanel();
-  } else {
-    openFilterPanel();
-  }
-});
+function Filter(elm) {
+  this.$elm = $(elm);
+  this.$input = this.$elm.find('input[name]');
+  this.$remove = this.$elm.find('.button--remove');
+
+  this.$input.on('change', this.handleChange.bind(this));
+  this.$remove.on('click', this.handleClear.bind(this));
 
-var prepareValue = function($elm, value) {
-  if ($elm.attr('type') === 'checkbox') {
-    return $.isArray(value) ? value : [value];
+  this.name = this.$input.eq(0).attr('name');
+  this.fields = [this.name];
+}
+
+Filter.build = function($elm) {
+  if ($elm.hasClass('date-choice-field')) {
+    return new DateFilter($elm);
+  } else if ($elm.hasClass('js-typeahead-filter')) {
+    return new TypeaheadFilter($elm);
   } else {
-    return value;
+    return new Filter($elm);
   }
-  return $elm.attr('type') === 'checkbox' ? [value] : value;
 };
 
-function ensureVisible($elm) {
-  if ($elm.is(':visible')) {
-    return;
-  }
-  var $accordion = $elm.closest('.js-accordion');
-  if ($accordion.length) {
-    accordion.showHeader($accordion.find('.js-accordion_header'));
+Filter.prototype.fromQuery = function(query) {
+  this.setValue(query[this.name]);
+  return this;
+};
+
+Filter.prototype.setValue = function(value) {
+  var $input = this.$input.data('temp') ?
+    this.$elm.find('#' + this.$input.data('temp')) :
+    this.$input;
+  $input.val(prepareValue($input, value)).change();
+  return this;
+};
+
+Filter.prototype.handleClear = function() {
+  this.setValue();
+  this.$input.focus();
+};
+
+Filter.prototype.handleKeydown = function(e) {
+  if (e.which === KEYCODE_ENTER) {
+    e.preventDefault();
+    this.$input.change();
```

Comment:
Not sure if it's caused by this function, but I noticed that when hitting enter on a regular text filter it triggers a form submit / page reload. Same with clicking "Apply filters". This does not seem to be the case on master.